### DISCUSSION TITLE
How `leading` parameter of `par` function works?

### DISCUSSION THREAD

**Discussion by PgLoLo** [typst/typst#4397] [Q&A] [answered]
Hello!

I need to have 1.5 spacing between lines in my document, but I got inconsisten results with `leading` parameter of `par` function. Specifically, spacings with this parameter are much larger then in word or latex. Consider the following example:

typst:
```typst
#set text(14pt)

#set par(leading: 1.5em)

#lorem(100)
```
![image](https://github.com/typst/typst/assets/5480856/8a99c933-c099-4947-96f8-0a75c20e45fc)

latex:
```tex
\documentclass[14pt]{extarticle}

\usepackage[a4paper, left=25mm, right=10mm, top=20mm, bottom=20mm]{geometry}

\usepackage{setspace}

\begin{document}

    \onehalfspacing

    Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magnam aliquam quaerat voluptatem. Ut enim aeque doleamus animo, cum corpore dolemus, fieri tamen permagna accessio potest, si aliquod aeternum et infinitum impendere malum nobis opinemur. Quod idem licet transferre in voluptatem, ut postea variari voluptas distinguique possit, augeri amplificarique non possit. At etiam Athenis, ut e patre audiebam facete et urbane Stoicos irridente, statua est in quo a nobis philosophia defensa et collaudata est, cum id, quod maxime placeat, facere possimus, omnis voluptas assumenda est, omnis dolor repellendus. Temporibus autem quibusdam et.

\end{document}
```
![image](https://github.com/typst/typst/assets/5480856/3c6cea26-3ff4-4821-83b7-5e8fd8f57be0)

Google docs:
![image](https://github.com/typst/typst/assets/5480856/ec7e5d6a-5be0-467d-8ec8-d06f608e5dfe)

Clearly, spacing in typst is much larger then spacing in latex or google docs and looks more like double spacing.

Thus my question is: what exactly is `leading` argument of `par` function and how can I achive 1.5 spacing between lines in document?

**Comment by Enivex** [comment, marked as answer]
The leading parameter measures the distance between the bottom-edge of a line and the top-edge of the next.

(Though the model may change in the future)

**Comment by jamesrswift** [comment]
For 1.5 spacing you probably want a leading of 1.2em

**Reply by PgLoLo** [reply]
Can you please elaborate how did you get 1.2em? If it measures the distance between the bottom-edge of a line and the top-edge of the next, as @Enivex said, then should not it be like 1em?

**Reply by laurmaedje** [reply]
See also: https://github.com/typst/typst/issues/106#issuecomment-2041051807
